marshmallow_pagination/schemas.py

```diff
@@ -14,16 +14,22 @@ class PageMeta(ma.schema.SchemaMeta):
     """Metaclass for `PageSchema` that creates a `Nested` field based on the
     options configured in `OPTIONS_CLASS`.
     """
-    def __new__(mcs, name, bases, attrs):
-        klass = super().__new__(mcs, name, bases, attrs)
-        opts = klass.OPTIONS_CLASS(klass.Meta)
-        klass._declared_fields[opts.results_field_name] = ma.fields.Nested(
+
+    @classmethod
+    def get_declared_fields(mcs, klass, *args, **kwargs):
+        declared_fields = kwargs.get('dict_class', dict)()
+        opts = klass.opts
+        declared_fields[opts.results_field_name] = ma.fields.Nested(
             opts.results_schema_class,
             attribute='results',
             many=True,
             **opts.results_schema_options
         )
-        return klass
+        base_fields = super(PageMeta, mcs).get_declared_fields(
+            klass, *args, **kwargs
+        )
+        declared_fields.update(base_fields)
```

Comment:
This is a subtle issue, but this logic causes a problem. We're using `PageMeta` as the metaclass of a few different abstract page schemas (e.g. `OffsetPageSchema`). This `get_declared_fields` method runs when we make the abstract schema, setting a `results` field with a `nested` value of `None`. That's fine for an abstract schema, but the problem is that schemas that subclass `OffsetPageSchema` now have their `results` fields overridden by the unusable `results` field on the base class. I'm going to switch this back to the way it was, where the metaclass overrides any existing `results` fields.

Anyway, my fault for not writing tests (yet).